semantic_conventions/metrics/container.yaml

```diff
@@ -0,0 +1,97 @@
+groups:
+  - id: metric.container.cpu.usage
+    type: metric
+    metric_name: container.cpu.usage
```

Comment:
is this needed separate from `system.cpu.usage`?

(there was some related discussion in https://github.com/open-telemetry/opentelemetry-specification/pull/2388)

Reply:
Thank's @trask! That's interesting one and it's quite similar to https://github.com/open-telemetry/opentelemetry-specification/pull/2388#issuecomment-1057120767.

I will try to share my view on this :).
So, from infrastructure's observability point of view, container metrics would be collected from outside the containers themselves (as a best practice). 
For example by using the Docker runtime's APIs or Kubelet's API's.

So the point here is that it does not make a lot of sense to treat a container as a host and collect its metrics by running a collector inside the container.

So for metrics collected through the runtime/orchastrator APIs I think we need to be specific and have a resource specific namespace like `container.cpu`,  `pod.cpu` etc.

In many cases the CPU/Memory resources are limited as well: 
- https://docs.docker.com/config/containers/resource_constraints/#cpu
- https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#specify-a-cpu-request-and-a-cpu-limit

In very specific usecases, if someone wants to treat a container as a host ([`kind`](https://kind.sigs.k8s.io/) uses containers' as hosts) then the collector should be running on the container/host directly and report metrics under the `system.*` namespace. In that case the resource is a "host" not a container from the observation/collection point of view.

Also related to https://github.com/open-telemetry/semantic-conventions/issues/226#issuecomment-1663565922.